### PR TITLE
feat(dashboard): composite API calls to list composite asset properties

### PR DESCRIPTION
**Pull request by ssjagad**
## Overview
To be able to list asset composite properties we want to call the new Sitewise APIs listAssetProperties and listAssetModelProperties. 
     - Changed the resource exploration flow to have an object that contains both assetId and assetModel
     - updated listModeledDataStreams to use new APIs to get result
     - Added Filter: 'ALL' parameter in model centric viz call for assetModelProperties to get composite models too
     - updated SDKs

https://github.com/awslabs/iot-app-kit/assets/145582655/43ad0d2b-54ce-433d-aedf-7965fcca6a59

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/types.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { AssetPropertySummary, AssetModelPropertySummary } from '@aws-sdk/client-iotsitewise';
+
+type listPropertyOutputExtender = {
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/types.ts:3]: Types should be written in pascal case, such as `ListPropertyOutputExtender`.

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/types.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export type newListAssetModelPropertiesCommandOutput = {
+  assetModelPropertySummaries: newAssetModelPropertySummary[] | undefined;
+};
+
+export type SelectedAssetProp = {
+  assetId: string;
+  assetModelId: string;
+};
+
+export type SelectedAssetsPropsArray = SelectedAssetProp[];
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/types.ts:30]: Can this just be `SelectedAsset`? Appending `Prop` to the end doesn't seem to add meaning and may actually add confusion.

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/useModeledDataStreams/useModeledDataStreams.ts [hunk 3]
```diff
@@ -4,19 +4,20 @@ import invariant from 'tiny-invariant';
...
   client: IoTSiteWiseClient;
 }
 
-export function useModeledDataStreams({ assetIds, client }: UseModeledDataStreamsProps) {
+export function useModeledDataStreams({ assetProps, client }: UseModeledDataStreamsProps) {
   const cacheKeyFactory = new ModeledDataStreamCacheKeyFactory();
   const queries =
     useQueries({
-      queries: assetIds.map((assetId) => ({
+      queries: assetProps.map((assetProperties) => ({
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/useModeledDataStreams/useModeledDataStreams.ts:18]: You called the item `assetProperties`, yet it does not appear we are iterating over asset properties.

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/types.ts (hunk 1 above)
Comment (tracy-french) [reply packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/types.ts:3]: Also, `Extender` sounds like it performs an action, yet this construct has no methods on it. What about something like `ExtendedAssetPropertySummaryProperties`?

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/types.ts (hunk 2 above)
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/types.ts:30]: Generally, you should avoid `Array` types in TypeScript. They don't offer much value. Instead, just annotate types as `SelectedAssetProp[]` as needed.

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/useModeledDataStreams/listModeledDataStreamsRequest.ts [hunk 4]
```diff
@@ -1,76 +1,127 @@
...
 import type { ModeledDataStream } from '../types';
 import { createNonNullableList } from '~/helpers/lists/createNonNullableList';
 import { DEFAULT_STRING } from '~/components/queryEditor/contants';
+import { Paginator } from '@aws-sdk/types';
+import {
+  SelectedAssetProp,
+  AssetModelPropertySummaryMap,
+  newListAssetPropertiesCommandOutput,
+  newAssetModelPropertySummary,
+} from '../../types';
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/useModeledDataStreams/listModeledDataStreamsRequest.ts:20]: Ensure you use type imports when importing types.

File: packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/useModeledDataStreams/useModeledDataStreams.ts [hunk 5]
```diff
@@ -29,12 +30,12 @@ export function useModeledDataStreams({ assetIds, client }: UseModeledDataStream
...
   return async function ({
-    queryKey: [{ assetId }],
+    queryKey: [assetProperties],
     signal,
   }: QueryFunctionContext<ReturnType<ModeledDataStreamCacheKeyFactory['create']>>) {
-    invariant(assetId, 'Expected assetId to be defined as required by the enabled flag.');
+    invariant(assetProperties, 'Expected assetProp to be defined as required by the enabled flag.');
 
-    const request = new ListModeledDataStreamsRequest({ assetId, client, signal });
+    const request = new ListModeledDataStreamsRequest({ assetProp: assetProperties.assetProp, client, signal });
```
Comment (tracy-french) [inline packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/useModeledDataStreams/useModeledDataStreams.ts:38]: By destructuring the queryKey in the function parameters, such as `queryKey: [{ assetProp }]`, you can change the request to `const request = new ListModeledDataStreamRequest({ assetProp, client, signal });` and make things a little cleaner. :)

**Review by tracy-french — CHANGES_REQUESTED** [review]
Looks good overall, @ssjagad. There are a few items we should address before merging this in, primarily with naming and TypeScript semantics/best practices.

**Review by corteggiano — APPROVED** [review]

**Review by chejimmy — DISMISSED** [review]

**Pull request merged**